Composer/packages/adaptive-form/src/components/Comment.tsx

```diff
@@ -0,0 +1,166 @@
+  const [isEditing, setIsEditing] = useState(!comment);
+  const [isMultiline, setIsMultiline] = useState((comment ?? '').length >= MAX_CHARS_FOR_SINGLE_LINE);
+  const [showMore, setShowMore] = useState(false);
+  const [needsTruncation, setNeedsTruncation] = useState(false);
+  const noteRef = useRef<HTMLParagraphElement | null>(null);
+  const textfieldRef = useRef<ITextField>(null);
+  const editingStateRef = useRef<boolean>(isEditing);
+
+  useEffect(() => {
+    const el = noteRef.current;
+    setNeedsTruncation(Boolean(el && el.scrollHeight > el.clientHeight));
+  }, []);
+
+  useEffect(() => {
+    // send focus to the text field if toggling from view mode to edit mode
+    if (!editingStateRef.current && isEditing) {
+      textfieldRef.current?.focus();
+    } else if (!isEditing) {
+      const el = noteRef.current;
+      setNeedsTruncation(Boolean(el && el.scrollHeight > el.clientHeight));
+    }
+    editingStateRef.current = isEditing;
+  }, [isEditing]);
+
+  const handleChange = useCallback(
+    (_e, val?: string) => {
+      onChange(val);
+      setIsMultiline(Boolean(val && val.length >= MAX_CHARS_FOR_SINGLE_LINE));
+    },
+    [onChange]
+  );
+
+  const handleBlur = useCallback(() => {
+    if (comment && comment.length > 0) {
+      setIsEditing(false);
+    }
+  }, [comment]);
+
+  const menuItems: IContextualMenuItem[] = [
+    {
+      key: 'edit',
+      name: formatMessage('Edit'),
+      iconProps: {
+        iconName: 'Edit',
+      },
+      onClick: () => {
+        setIsEditing(true);
+      },
+    },
+    {
+      key: 'delete',
+      name: formatMessage('Delete'),
+      iconProps: {
+        iconName: 'Delete',
+      },
+      onClick: () => {
+        handleChange({}, '');
+        setIsEditing(true);
+      },
+    },
+  ];
+
+  return (
+    <React.Fragment>
+      <TextField
+        componentRef={textfieldRef}
+        multiline={isMultiline}
+        placeholder={formatMessage('Add a note')}
+        resizable={false}
+        rows={5}
+        styles={{
+          root: {
+            display: isEditing ? undefined : 'none',
+          },
+          field: {
+            paddingRight: isMultiline ? '28px' : undefined,
+            paddingBottom: isMultiline ? '30px' : undefined,
+          },
+        }}
+        value={comment}
+        onBlur={handleBlur}
+        onChange={handleChange}
+      />
+      <div css={styles.note(isEditing)} data-testid="CommentCard">
+        <div css={styles.menu}>
+          <IconMenu
+            iconName="MoreVertical"
+            iconSize={12}
+            label={formatMessage('Comment menu')}
+            menuItems={menuItems}
+            menuWidth={120}
+          />
+        </div>
+        <p ref={noteRef} css={styles.noteBody(!showMore)}>
+          {comment}
+        </p>
+        {needsTruncation && (
+          <div css={styles.showMore}>
+            <Link styles={{ root: { fontSize: '12px' } }} onClick={() => setShowMore((current) => !current)}>
```

Comment:
a button here is a better choice for accessibility reasons (as="button" on Link is doable too)

Reply:
The Link component renders as a `button` if no `href` is passed.